cpp/include/kvikio/defaults.hpp

```diff
@@ -57,7 +57,13 @@ inline bool getenv_or(std::string_view env_var_name, bool default_val)
   }
   // Convert to lowercase
   std::string str{env_val};
-  std::transform(str.begin(), str.end(), str.begin(), ::tolower);
+  // Special considerations regarding the case conversion:
+  // - std::tolower() is not an addressable function. Passing it to std::transform() as
+  //   a function pointer causes the program behavior "unspecified (possibly ill-formed)",
+  //   hence the lambda.
+  // - To avoid UB in std::tolower(), the character must be cast to unsigned char.
+  std::transform(
+    str.begin(), str.end(), str.begin(), [](unsigned char c) { return std::tolower(c); });
```

Comment:
wow, good catch!

Reply:
Is there an advantage of using `::tolower` over `std::tolower`? From the references, they seem functionally the same. I guess perhaps the reason is passing `::tolower` would make the initial code compile, while `std::tolower` doesn't (instead the template version `std::tolower(int, const std::locale&)` is picked, which doesn't compile due to two parameters).